indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -1240,6 +1241,28 @@ Map<String, NavigableMap<DateTime, SortedMap<Interval, List<TaskLockPosse>>>> ge
     return running;
   }
 
+  public Set<TaskLock> getAllReplaceLocksForDatasource(final String datasource)
+  {
+    giant.lock();
+    try {
+      final NavigableMap<DateTime, SortedMap<Interval, List<TaskLockPosse>>> dsRunning = running.get(datasource);
```

Comment:
Maybe rename `dsRunning` to `activeLocksForDatasource` or just `activeLocks`

Reply:
Done